MaskedSelect.lua

```diff
@@ -0,0 +1,41 @@
+local MaskedSelect, parent = torch.class('nnd.MaskedSelect', 'nn.Module')
+
+--[[ Sets the provided mask value for the module. ]]
+function MaskedSelect:__init()
+  -- mask must be a tensor, convert to byte tensor
+  parent.__init(self)
+end
+
+--[[ Performs maskedSelect operation. ]]
+function MaskedSelect:updateOutput(input)
+  local input, mask = unpack(input)
+  self.output = input:maskedSelect(mask)
```

Comment:
This will allocate a new tensor each time. Re-use `self.output` instead:

`self.output:maskedSelect(input, mask)`